File: src/main/java/com/mbayou/kick4k/events/type/LivestreamMetadataUpdatedEvent.java
package com.mbayou.kick4k.events.type;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.time.Instant;

public class LivestreamMetadataUpdatedEvent extends KickEvent {
    private final EventUser broadcaster;
    private final LivestreamMetadata metadata;

    @JsonCreator
    public LivestreamMetadataUpdatedEvent(@JsonProperty("broadcaster") EventUser broadcaster,
                                          @JsonProperty("metadata") LivestreamMetadata metadata) {
        this.broadcaster = broadcaster;
        this.metadata = metadata;
    }

    public EventUser getBroadcaster() {
        return broadcaster;
    }

    public LivestreamMetadata getMetadata() {
        return metadata;
    }

    public static String getEventType() {
        return "livestream.metadata.updated";
    }

    public static String getEventVersion() {
        return "1";
    }
}

File: src/main/kotlin/com/mbayou/kick4k/events/type/LivestreamMetadataUpdatedEvent.kt
package com.mbayou.kick4k.events.type

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class LivestreamMetadataUpdatedEvent @JsonCreator constructor(
    @JsonProperty("broadcaster") val broadcaster: EventUser,
    @JsonProperty("metadata") val metadata: LivestreamMetadata,
) : KickEvent() {
    companion object {
        @JvmStatic
        fun getEventType(): String = "livestream.metadata.updated"

        @JvmStatic
        fun getEventVersion(): String = "1"
    }
}
